tests/job.go

```diff
@@ -10,37 +10,88 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"kubevirt.io/client-go/kubecli"
+)
 
-	. "github.com/onsi/gomega"
+const (
+	toSucceed = true
+	toFail    = false
 )
 
-func WaitForJobToSucceed(virtClient *kubecli.KubevirtClient, job *batchv1.Job, timeoutSec time.Duration) {
-	EventuallyWithOffset(1, func() bool {
-		job, err := (*virtClient).BatchV1().Jobs(job.Namespace).Get(job.Name, metav1.GetOptions{})
-		ExpectWithOffset(1, err).ToNot(HaveOccurred())
-		Expect(job.Status.Failed).NotTo(Equal(*job.Spec.BackoffLimit), "Job was expected to succeed but failed")
-		return job.Status.Succeeded > 0
-	}, timeoutSec*time.Second, 1*time.Second).Should(BeTrue(), "Job should succeed")
+// WaitForJobToSucceed blocks until the given job finishes.
+// On success, it returns with a nil error, on failure or timeout it returns with an error.
+func WaitForJobToSucceed(job *batchv1.Job, timeout time.Duration) error {
+	return waitForJob(job, toSucceed, timeout)
+}
 
+// WaitForJobToFail blocks until the given job finishes.
+// On failure, it returns with a nil error, on success or timeout it returns with an error.
+func WaitForJobToFail(job *batchv1.Job, timeout time.Duration) error {
+	return waitForJob(job, toFail, timeout)
 }
 
-func WaitForJobToFail(virtClient *kubecli.KubevirtClient, job *batchv1.Job, timeoutSec time.Duration) {
-	EventuallyWithOffset(1, func() int32 {
-		job, err := (*virtClient).BatchV1().Jobs(job.Namespace).Get(job.Name, metav1.GetOptions{})
-		ExpectWithOffset(1, err).ToNot(HaveOccurred())
-		Expect(job.Status.Succeeded).ToNot(BeNumerically(">", 0), "Job should not succeed")
-		return job.Status.Failed
-	}, timeoutSec*time.Second, 1*time.Second).Should(BeNumerically(">", 0), "Job should fail")
+func waitForJob(job *batchv1.Job, toSucceed bool, timeout time.Duration) error {
+	virtClient, err := kubecli.GetKubevirtClient()
+	if err != nil {
+		return err
+	}
+
+	jobCompleteError := func(job *batchv1.Job) error { return nil }
+	jobFailedError := func(job *batchv1.Job) error { return nil }
+	if toSucceed {
```

Comment:
Can this be simplified to
 ```
jobFailedError = func(job *batchv1.Job) error {
                        if toSucceed{
			         return fmt.Errorf("Job %s finished with failure, status: %+v", job.Name, job.Status)
                         }
                         return nil
		}
```

Reply:
Not sure if it is a simplification, but I am fine with your proposal.
(the cost of asking twice is negligible)

Reply:
Done